python/tchannel/tornado/dispatch.py

```diff
@@ -78,6 +76,13 @@ def handle_call(self, request, connection):
             request.endpoint += chunk
             chunk = yield request.argstreams[0].read()
 
+        # event: receive_request
+        if connection.tchannel:
+            request.tracing.name = request.endpoint
+            connection.tchannel.event_emitter.fire(
+                EventType.receive_request, request
```

Comment:
when we initialize TChannelProxy, we need to pass tchannel. I understand your concern, currently I haven't found a better to do that. We can add a TODO here to track this issue.